### ISSUE TITLE
From alpha-30 I get error Can't reexport the named export 'valueConverter' from non EcmaScript module (only default export is available)

### ISSUE THREAD

**Issue by magnusdanielson** [aurelia/aurelia#1498]
# 🐛 Bug Report

From alpha-30 I get the error when the webpack task runs

`Can't reexport the named export 'valueConverter' from non EcmaScript module (only default export is available)`

And many more for other Aurelia types.

## 🤔 Expected Behavior

Webpack should be able to build the bundle

## 😯 Current Behavior

The full error for one of the types is below

`Can't reexport the named export 'valueConverter' from non EcmaScript module (only default export is available)
    at HarmonyExportImportedSpecifierDependency._getErrors (C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\dependencies\HarmonyExportImportedSpecifierDependency.js:374:6)
    at HarmonyExportImportedSpecifierDependency.getErrors (C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\dependencies\HarmonyExportImportedSpecifierDependency.js:354:16)
    at Compilation.reportDependencyErrorsAndWarnings (C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\Compilation.js:1463:22)
    at C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\Compilation.js:1258:10
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\GitProjects\h2o\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
    at AsyncSeriesHook.lazyCompileHook (C:\GitProjects\h2o\node_modules\tapable\lib\Hook.js:154:20)
    at Compilation.finish (C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\Compilation.js:1253:28)
    at C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\Compiler.js:672:17
    at eval (eval at create (C:\GitProjects\h2o\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:11:1)
    at C:\GitProjects\h2o\node_modules\@microsoft\sp-build-web\node_modules\webpack\lib\Compilation.js:1185:12
 @ ./lib/webparts/helloWorld/HelloWorldWebPart.js`

## 💁 Possible Solution

None, so far.

## 🔦 Context

The end goal is to run Aurelia v2 in SPFx
https://docs.microsoft.com/en-us/sharepoint/dev/spfx/sharepoint-framework-overview

## 💻 Code Sample

This repo hold reference to alpha-29 and works. https://github.com/magnusdanielson/spfx-au2/tree/spfx1.15

If you upgrade @aurelia/plugin-gulp and aurelia to anything above alpha-29 it breaks with many error similar to above.

The example also holds reference to https://github.com/n8design/htwoo but that is irrelevant for the issue.

## 🌍 Your Environment

I am building solution with SPFx that uses typescript and webpack. In this case I have also included aurelia-gulp to make it work. I could not get the aurelia-webpack plugin to work properly.

Node version 14.9.0
Npm version 6.14.16
Webpack 4.4.46
Typescript 4.5

| Software         | Version(s) |
| ---------------- | ---------- |
| Aurelia          | alpha-30
| Language         | typescript 4.5
| Browser          | Chrome-any version
| Bundler          | gulp+webpack
| Operating System | windows
| NPM/Node/Yarn    | npm

**Comment by bigopon** [comment]
@magnusdanielson thanks for reporting the issue. I assume that you have tried with the latest version alpha 37?

**Comment by magnusdanielson** [comment]
Yes, same error with alpha 37.

**Comment by magnusdanielson** [comment]
Ok, I fixed and it now works with alpha 37.

I had to add below rules to webpack.
`{
        include: /node_modules/,
        test: /\.mjs$/,
        type: 'javascript/auto'
 }`

**Comment by magnusdanielson** [comment]
A bit to early to close.

Now I get a problem with DI.

```
Error: AUR0012:IContainer
    at Container.R (index.cjs:1051:10)
    at Container.get (index.cjs:933:1)
    at Container.ft (index.cjs:763:1)
    at Array.map (<anonymous>)
    at Factory.construct (index.cjs:778:1)
    at Resolver.resolve (index.mjs:715:1)
    at Container.get (index.cjs:934:1)
    at new Controller (index.cjs:2959:1)
    at Controller.$el (index.cjs:3008:1)
    at index.cjs:3739:1
```

It looks like it tries to resolve IContainer.

![image](https://user-images.githubusercontent.com/12081481/185141391-55830d7a-df67-41c4-a66d-1a95dc563847.png)

This is my start up

```
var au = new Aurelia();
      const rootContainer = au.container;
      rootContainer.register(
        Registration.instance("WebPartContext", this.context),
        Registration.instance("WebPartProperties", this.properties)
      );
      this.element = document.getElementsByClassName(styles.helloWorld)[0];

      await au.register(<any>HelloWorldMyComponent)
        .register(<any>HelloWorldOtherStuff)
        .app({
          component: HelloWorldMyComponent,
          host: this.element
        })
        .start();
```

And this is the HelloWorldMyComponent that this.element is pointing to

```
@inject(Element, "WebPartContext","WebPartProperties")
export class HelloWorldMyComponent
{
    constructor(private element:Element, private context:WebPartContext, private properties:IHelloWorldWebPartProps )
    {
```

**Comment by bigopon** [comment]
@magnusdanielson it seems you are not injecting Icontainer anywhere so its probably a module duplication issue because of the upgrade. Can you try removing both node modules and lock file then reinstalling the deps?

**Comment by magnusdanielson** [comment]
First of all, thank you for trying to help me out @bigopon.

Not much luck unfortunately. Still the same problem.

Som objects do get resolved from the container before the issue occurs. I don't understand why it tries to resolve IContainer from the container, though. It seams a bit meta to me, but I probably just get that wrong.

The following gets resolved properly:

- IWorkTracker
- ILogger
- ILogConfig
- ILogEventFactory
- IRendering

with what I understand the below container.

![image](https://user-images.githubusercontent.com/12081481/185867304-e1f81cfe-94c0-4d1d-884a-b6a89211c727.png)

I also tried the advanced low level setup https://docs.aurelia.io/developer-guides/cheat-sheet#advanced-low-level but it has the same problem.

**Comment by bigopon** [comment]
@magnusdanielson It seems quite hard to figure out where to start looking for issue. Can you help create a small repro based on this https://stackblitz.com/edit/typescript-jzw4qu?file=app.html

**Comment by magnusdanielson** [comment]
@bigopon The build setup is a bit complicated and out of my control. I don't think it is possible to modify the build on stackblitz so it mimics the setup Microsoft has made for SPFx.

The repo here https://github.com/magnusdanielson/spfx-au2/tree/spfx1.15 is updated with the last "fix" above to allow for mjs files in webpack.

If pulling the repo does not work for you, would it help if I created a docker container where everything is included?

The end goal is to start with Microsoft SPFx setup and then add or change what is needed to get Aurelia2 to work. I have it all working fine with alpha-29. I have tried to look at changes between 29 and 30 to understand what could cause the problem but it is a bit hard to follow along.

**Comment by bigopon** [comment]
I tried running the build script and it failed:
![image](https://user-images.githubusercontent.com/9994529/186892214-1b27f67b-b9ed-4d1d-8b98-714be06012b3.png)

Beside that, it's quite hard to figure out the reproduction step of the issue. I think there'll need to be more info on this.

The components look pretty simple to me, shouldn't you be able to replicate it in a different, simpler setup?

**Comment by Vheissu** [comment]
@magnusdanielson How did you go with this issue, any luck or further details?

**Comment by magnusdanielson** [comment]
@Vheissu sorry guys. I got swamped with work and illness. I will take a look at this again as soon as possible. I will not give up. I will make this work. The error @bigopon is referring to is because i have a mistake in the tsconfig file. It is one of the esnext settings below that makes webpack angry.

````
"target": "esnext",
"module": "esnext",
````

Let me take a fresh attack on this and I will get back to you.

Once again, really cool framework you have made. The UI in this Microsoft Teams app is made by my team with Aurelia. https://www.livearena.com/

**Comment by bigopon** [comment]
This shouldn't happen anymore, so let's close this for now and if there's still any similar error, please open a new issue.
